src/poller/popol.rs

```diff
@@ -61,6 +64,8 @@ impl Poller {
 }
 
 impl Poll for Poller {
+    type Waker = UnixStream;
```

Comment:
Why not use `popol::Waker`?

Reply:
Or you could create a newtype `struct Waker(Arc<popol::Waker>)`

Reply:
Why would I need `Arc` here?

Reply:
For `Clone`

Reply:
Ok, `popol::Waker` requires to be provided with `Sources` object in the `Waker::new` constructor, which doesn't fit the current architecture here... I can't add this parameter to my `Waker` trait object since it will not fit with other epoll providers.

Reply:
The latest version doesn't require that anymore: https://github.com/cloudhead/popol/commit/a8819a80351f9df55bcdbbd9183a42e053bd7092#diff-b1a35a68f14e696205874893c07fd24fdb88882b47c23cc0e0c80a30c7d53759R496

Reply:
Going to push a new release with that change.

Reply:
Ok, `popol-3.0.0` has the new API.

Reply:
Ok, with the new `popol::Waker` version I still have a holistic object, while what I need is an object splittable into two parts. Of course I can do `Arc` and clone, but this will make API in reactor more cluttered - and will also unnecessary ship sending part to the reactor and listening part to the service. So why do you think that use of `popol::Waker` has an advantage over the current approach?

Reply:
Hmm I understand the issue, though I think they are fairly minor and can be worked around using the `WakerSend` and `WakerRecv` traits to hide the underlying capabilities. Later on if we see a big advantage to splitting the types, it's pretty easy to add a `split` method to `popol::Waker`.

I think using `popol::Waker` has the advantage that you don't have to maintain that code in this library, since it's already written, and if `popol` changes the way it works, it'll just ship an updated waker and won't break. If on the other hand popol changes its waker but you are using a custom one here, there could be an incompatibility.

Reply:
I agree with you - but I wish it was that easy...

If I do it like
```rust
impl Waker for popol::Waker {
    type Send = Arc<popol::Waker>;
    type Recv = Arc<popol::Waker>;

    fn pair() -> Result<(Self::Send, Self::Recv), Error> {
        let waker = Arc::new(popol::Waker::new()?);
        Ok((waker.clone(), waker))
    }
}
```

I can't do `impl io::Read for Arc<popol::Waker>` etc. Without Arc I can't split waker into two parts.

Reply:
Ok, I finally got it. Still not linear - pls check the last commit where I did this.